sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -62,7 +75,10 @@ object IntegerLiteral {
   }
 }
 
-case class Literal(value: Any, dataType: DataType) extends LeafExpression {
+case class Literal (var value: Any, dataType: DataType) extends LeafExpression {
```

Comment:
There are 400-500 places using this constructor, I'd like to leave the refactor out for another PR.